webapp/src/main/java/com/box/l10n/mojito/quartz/QuartzPollableTaskScheduler.java

```diff
@@ -161,6 +172,34 @@ public <I, O> PollableFuture<O> scheduleJob(QuartzJobInfo<I, O> quartzJobInfo) {
     return new QuartzPollableFutureTask<O>(pollableTask, jobOutputType);
   }
 
+  private <I, O> QuartzPollableFutureTask<O> getPendingPollableFutureIfExists(
+      QuartzJobInfo<I, O> quartzJobInfo, Scheduler scheduler) {
+    logger.debug(
+        "DisallowConcurrentExecution is set, checking if the job already has a trigger in a blocked state");
+    String jobKeyName = getKeyName(quartzJobInfo.getClazz(), quartzJobInfo.getUniqueId());
+    try {
+      List<Trigger> triggers =
+          (List<Trigger>) scheduler.getTriggersOfJob(new JobKey(jobKeyName, DYNAMIC_GROUP_NAME));
```

Comment:
how many triggers can be returned here?

Reply:
My understanding is that for a job annotated with DisallowConcurrentExecution where a re-schedule occurs there will be a max of two, one in the COMPLETED state (which is the trigger that fired the currently executing job), and one in a BLOCKED state which is waiting to execute when the other job finishes.